pkg/task/restore.go

```diff
@@ -420,10 +418,14 @@ func dropToBlackhole(
 func filterRestoreFiles(
 	client *restore.Client,
 	cfg *RestoreConfig,
-) (files []*backuppb.File, tables []*utils.Table, dbs []*utils.Database) {
+) (files []*backuppb.File, tables []*utils.Table, dbs []*utils.Database, filledTables []*utils.Table) {
 	for _, db := range client.GetDatabases() {
 		createdDatabase := false
 		for _, table := range db.Tables {
+			if table.TotalBytes > 0 {
+				filledTables = append(filledTables, table)
+			}
```

Comment:
um, so,

when we perform restore, there are two "tables" to deal with:

1. the table inside the backup archive, or the "source"
2. the live table in the target database, or the "destination"

what BR restore does is copy all data from the "source" into the "destination". 

**here, `table.TotalBytes` refer to the size of the "source", not the size of "destination"**.

in #801 the meaning of "table has data" refers to the destination, i.e. checking if the table in the target database is empty. 

if the destination is not empty, we cannot ensure `checksum(source) == checksum(destination)`.

i suggest you do a simple check before submitting a fix:

1. BR backup a table `t` into `local:///tmp/sample/`
2. Do not truncate `t`, keep it non-empty.
3. BR restore from `local:///tmp/sample/`
4. If #801 is active, it should throw a warning/error because table `t` is not empty.
5. Now truncate table `t`
6. BR restore from `local:///tmp/sample/` again.
7. There should be no errors, and table `t` is filled with the old data.

Reply:
I am extremely sorry for misunderstanding the issue completely. Thank you again for taking your time and clarifying the issue in detail, indeed they were different tables I was looking at. For the steps you mentioned I tried them out and there seems to be no error whatsoever in any step, so I guess the issue was fixed already. With your go I will close this PR hence worth.

Reply:
I think this issue is not fixed yet. 
Before step.3. you can try to insert some data into table. `t`. and step.4. will throw an error which is `checksum mismatch...`.